### PR TITLE
Add commands to print Action and ActionResult data from text proto

### PR DESCRIPTION
**Pull request by cdlee2**
This adds printing the information from the Action and ActionResult protos. For printing actions, the Command and input roots are grabbed from cache and displayed in human readable format alongside the contents of the Action proto. Similarly, output directories in the given ActionResult are listed recursively.

### CODE REVIEW

File: src/main/java/com/google/devtools/build/remote/client/RemoteClient.java [hunk 1]
```diff
@@ -14,35 +14,64 @@
...
 
-  AbstractRemoteActionCache cache;
-  DigestUtil digestUtil;
+  private final AbstractRemoteActionCache cache;
+  private final DigestUtil digestUtil;
+
+  private static final Joiner SPACE_JOINER = Joiner.on(' ');
+  private static final Escaper STRONGQUOTE_ESCAPER =
+      new CharEscaperBuilder().addEscape('\'', "'\\''").toEscaper();
+  private static final CharMatcher SAFECHAR_MATCHER =
```
Comment (agoulti) [inline src/main/java/com/google/devtools/build/remote/client/RemoteClient.java]: Would it make sense to move this into a separate class, the way Bazel's one works?
It is quite likely we'll need Windows support and the way they've set up their abstraction seems reasonable (might also be easier to keep those in sync)

Comment (cdlee2) [reply src/main/java/com/google/devtools/build/remote/client/RemoteClient.java]: I think that's a good idea. I'll write some tests too.

A note on Windows support: While there is a Windows command describer like the Linux one in Bazel [here](https://source.bazel.build/bazel/+/master:src/main/java/com/google/devtools/build/lib/util/CommandFailureUtils.java;l=88?q=CommandFailureUtils.java), escaping the command appears to have been a TODO for a while (2010). Windows will probably also require other command formatting work like changing forward slashes to back slashes.

Comment (agoulti) [reply src/main/java/com/google/devtools/build/remote/client/RemoteClient.java]: Yes, creating Windows support is not currently part of our mandate - but let's position ourselves to benefit from changes when they do occur in Bazel.

Comment (cdlee2) [reply src/main/java/com/google/devtools/build/remote/client/RemoteClient.java]: Done.

Comment (cdlee2) [reply src/main/java/com/google/devtools/build/remote/client/RemoteClient.java]: Done.

**Comment by cdlee2** [comment]
@ola-rozenfeld friendly ping?

File: src/main/java/com/google/devtools/build/remote/client/RemoteClient.java [hunk 2]
```diff
@@ -118,14 +135,130 @@ private void listOutputDirectory(OutputDirectory dir, int limit) throws IOExcept
...
+  private void listTree(Path path, Tree tree, int limit) throws IOException {
     Map<Digest, Directory> childrenMap = new HashMap<>();
     for (Directory child : tree.getChildrenList()) {
       childrenMap.put(digestUtil.compute(child), child);
     }
     listDirectory(path, tree.getRoot(), childrenMap, limit);
   }
 
+  private static int getNumFiles(Tree tree) {
+    return tree.getChildrenList().stream().mapToInt(dir -> dir.getFilesCount()).sum();
```
Comment (ola-rozenfeld) [inline src/main/java/com/google/devtools/build/remote/client/RemoteClient.java:147]: Nice! I just learned this syntax a few days ago, it's really cool! Java getting closer and closer to Python :-)

File: src/main/java/com/google/devtools/build/remote/client/RemoteClient.java [hunk 3]
```diff
@@ -118,14 +135,130 @@ private void listOutputDirectory(OutputDirectory dir, int limit) throws IOExcept
...
+
+  // Display output file (either digest or raw bytes).
+  private void printOutputFile(OutputFile file, boolean showRawOutputs) {
+    String contentString;
+    if (file.hasDigest()) {
+      contentString = "Content digest: " + digestUtil.toString(file.getDigest());
+    } else if (showRawOutputs) {
+      contentString =
+          String.format(
+              "Raw contents: '%s', size (bytes): %d",
```
Comment (ola-rozenfeld) [inline src/main/java/com/google/devtools/build/remote/client/RemoteClient.java:210]: We could consider having something like --show_contents_size_limit=20000 which will show everything less than the provided value of bytes. In the next CL, though.

**Review by ola-rozenfeld — APPROVED** [review]

File: src/main/java/com/google/devtools/build/remote/client/RemoteClient.java (hunk 3 above)
Comment (cdlee2) [reply src/main/java/com/google/devtools/build/remote/client/RemoteClient.java:210]: sounds good to me.

**Review by agoulti — APPROVED** [review]

**Pull request merged**